### PR TITLE
Heat exchanger bypass

### PR DESCRIPTION
**Pull request by FJanssen-TNO**
Allowing for bypassing of the heat exchanger when both the supply and the return on one side of the heat exchanger (primary or secondary) have the same temperature. This allows for flow through the heat exchanger without heat exchange, in a cascaded network where the temperature has dropped.

### CODE REVIEW

**Comment by KobusVanRooyen** [comment]
@FJanssen-TNO review has been done

File: src/mesido/esdl/esdl_heat_model.py [hunk 1]
```diff
@@ -917,14 +917,14 @@ def convert_heat_exchanger(self, asset: Asset) -> Tuple[Type[HeatExchanger], MOD
                 f"{params_t['Secondary']['T_supply']}. This is not possible as the HEX can only "
                 "transfer milp from primary to secondary."
             )
-            assert params_t["Primary"]["T_supply"] >= params_t["Secondary"]["T_supply"]
+            # assert params_t["Primary"]["T_supply"] >= params_t["Secondary"]["T_supply"]
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/esdl/esdl_heat_model.py:920]: I think so, but I want you to agree. This assert only makes sense when we don't have a bypass situation. But maybe I should add a potential error here, that can be extracted in the grow workflow when we don't allow for bypasses. Please drop your thoughts.

File: src/mesido/esdl/esdl_heat_model.py [hunk 2]
```diff
@@ -917,14 +917,14 @@ def convert_heat_exchanger(self, asset: Asset) -> Tuple[Type[HeatExchanger], MOD
...
         if params_t["Primary"]["T_return"] < params_t["Secondary"]["T_return"]:
             logger.error(
                 f"{asset.name} has a primary side return temperature that is lower than the "
                 f"secondary return temperature. This is not possible as the HEX can only transfer "
                 f"milp from primary to secondary."
             )
-            assert params_t["Primary"]["T_return"] >= params_t["Secondary"]["T_return"]
+            # assert params_t["Primary"]["T_return"] >= params_t["Secondary"]["T_return"]
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/esdl/esdl_heat_model.py:927]: See above

File: src/mesido/esdl/esdl_heat_model.py [hunk 3]
```diff
@@ -936,12 +936,15 @@ def convert_heat_exchanger(self, asset: Asset) -> Tuple[Type[HeatExchanger], MOD
...
+                    asset.attributes["capacity"] if asset.attributes["capacity"] else math.inf
+                )
 
-        max_heat_transport = (
-            params_t["Primary"]["T_supply"]
-            * max_power
-            / (params_t["Primary"]["T_supply"] - params_t["Primary"]["T_return"])
-        )
+        dt_prim = params_t["Primary"]["T_supply"] - params_t["Primary"]["T_return"]
+        dt_prim = dt_prim if dt_prim > 0.0 else 10
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/esdl/esdl_heat_model.py:945]: I've updated it with a comment and ensured same value for both primary and secondary is used.

File: src/mesido/esdl/esdl_heat_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KobusVanRooyen) [reply src/mesido/esdl/esdl_heat_model.py:920]: I agree. Add it to the potential error variable

**Pull request merged**